apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/backend/ir/FieldMergeUtils.kt

```diff
@@ -0,0 +1,64 @@
+package com.apollographql.apollo.compiler.backend.ir
+
+internal object FieldMergeUtils {
+
+  fun List<BackendIr.Field>.mergeFields(otherFields: List<BackendIr.Field>): List<BackendIr.Field> {
+    val fieldsToAdd = otherFields.toMutableList()
+    return this.map { field ->
+      val fieldToMergeIndex = fieldsToAdd.indexOfFirst { otherField -> otherField.responseName == field.responseName }
+      val fieldToMerge = if (fieldToMergeIndex >= 0) fieldsToAdd.removeAt(fieldToMergeIndex) else null
+      if (fieldToMerge == null) {
+        field
+      } else {
+        field.merge(fieldToMerge)
+      }
+    } + fieldsToAdd
+  }
+
+  private fun BackendIr.Field.merge(otherField: BackendIr.Field): BackendIr.Field {
+    val mergedFields = this.fields.mergeFields(otherField.fields)
+    return this.copy(
+        args = (this.args + otherField.args).distinct(),
```

Comment:
This feels weird. I'd expect the arguments to be the same at this point. Should we throw if they're not?